### PR TITLE
US49481-Included handling of optional VNI and VID for network creation

### PR DESCRIPTION
**Pull request by dbozzato81**
Please refer to GL Metal https://github.com/hpe-hcss/quake/pull/2931

### CODE REVIEW

File: v1/api/swagger/components/schemas/LocationInfo.yaml [hunk 1]
```diff
@@ -22,3 +24,9 @@ properties:
     type: string
     example: FTC1
     description: Data center descriptor
+  PrivateNetworkVIDRequired:
+    type: boolean
```
[the remark this answers is not in the document]
Comment (dbozzato81) [reply v1/api/swagger/components/schemas/LocationInfo.yaml:28]: Yes both fields are optional. I'll switch to PrivateNetworkVIDSupport and PrivateNetworkVNISupport

**Review by chaitra-mylarappachar — APPROVED** [review]

File: v1/api/swagger/components/schemas/NewNetwork.yaml [hunk 2]
```diff
@@ -23,3 +23,9 @@ properties:
       If not specified, the network uses the default IP pool allocated from the 
       hoster/project IP Pool factory and if specified, a new custom IP Pool will 
       be created and used for IP allocations.
+  VLAN:
+    type: integer
+    description: VLAN ID of the network
```
Comment (mchuang3) [inline v1/api/swagger/components/schemas/NewNetwork.yaml:28]: Is this used for Create only, or for both Create & Get? If it is for Create only, make it more verbose like the following. If it is used for Get as well, then what you have is good enough. Same for VNI below.
```suggestion
    description: optional VLAN ID of the network. If not specified, it is allocated from reserved pool.
```

File: v1/api/swagger/components/schemas/NewNetwork.yaml [hunk 3]
```diff
@@ -23,3 +23,9 @@ properties:
...
+  VNI:
+    type: integer
+    description: VNI ID of the network
```
Comment (mchuang3) [inline v1/api/swagger/components/schemas/NewNetwork.yaml:31]:
```suggestion
    description: Optional VNI ID of the network. If not specified, it is allocated from reserved pool if required.
```

**Review by mchuang3 — COMMENTED** [review]
Please add reference to the quake PR# in this PR description. Thanks.

**Review by mchuang3 — APPROVED** [review]

**Pull request merged**